internal/functions/efgs/upload-batch.go

```diff
@@ -216,7 +216,7 @@ func handleErrorUploadResponse(ctx context.Context, resp *efgsapi.UploadBatchRes
 
 		// This does binary search.
 		pos := sort.Search(len(batchDbKeys), func(i int) bool {
-			return bytes.Compare(batchDbKeys[i].KeyData, lookFor) <= 0
+			return strings.Compare(batchDbKeys[i].KeyData, string(lookFor)) <= 0
```

Comment:
1. Use base64 encoding instead of bytes->string conversion
2. Don't do this in a loop, do it right after the `lookFoor` declaration